src/operation/operation_model_interface.jl

```diff
@@ -109,23 +109,31 @@ function build_impl!(model::OperationModel, serialize::Bool)
     return get_status(model)
 end
 
-function _pre_solve_model_checks(model::OperationModel, optimizer; kwargs...)
+function build_if_not_already_built!(model; kwargs...)
     if !is_built(model)
         if !haskey(kwargs, :output_dir)
             error(
                 "'output_dir' must be provided as a kwarg if the model build status is $(get_status(model))",
             )
         else
-            status = build!(model; kwargs...)
+            new_kwargs = Dict(k => v for (k, v) in kwargs if k != :optimizer)
+            status = build!(model; new_kwargs...)
             if status != BuildStatus.BUILT
                 error("build! of the $(typeof(model)) failed: $status")
             end
         end
     end
+end
+
+function _pre_solve_model_checks(model::OperationModel, optimizer)
     jump_model = get_jump_model(model)
     if optimizer !== nothing
         JuMP.set_optimizer(jump_model, optimizer)
     end
+
+    optimizer_name = JuMP.solver_name(jump_model)
+    @info "Solving $(typeof(model)) with optimizer = $optimizer_name"
```

Comment:
We could potentially call `JuMP.backend(jump_model)` to capture more information.

Reply:
Yes, this is correct. We could add that info this PR to capture solver version also.